pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -116,7 +120,7 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 	totalTargets := len(targets)
 
 	// health check all targets and reconcile mhc status
-	currentHealthy, needRemediationTargets, nextCheckTimes, errList := healthCheckTargets(targets)
+	currentHealthy, needRemediationTargets, nextCheckTimes, errList := healthCheckTargets(targets, r.recorder)
```

Comment:
small nit: can we instead make r the receiver inline with the other funcs needing r

Reply:
fixed